diff --git a/alacritty/src/main.rs b/alacritty/src/main.rs
index eff1a0a2..a15b7edc 100644
--- a/alacritty/src/main.rs
+++ b/alacritty/src/main.rs
@@ -157,6 +157,17 @@ fn alacritty(mut options: Options) -> Result<(), Box<dyn Error>> {
     #[cfg(not(any(feature = "x11", target_os = "macos", windows)))]
     info!("Running on Wayland");
 
+    // HORRIBLE HACK DO NOT GET INSPIRED BY IT!
+    if env::var_os("APPDIR").is_some() {
+        env::remove_var("APPDIR");
+        env::remove_var("GSETTINGS_SCHEMA_DIR");
+        env::remove_var("LIBGL_DRIVERS_PATH");
+        env::remove_var("LIBVA_DRIVERS_PATH");
+        env::remove_var("GBM_BACKENDS_PATH");
+        env::remove_var("__EGL_VENDOR_LIBRARY_DIRS");
+        env::remove_var("GCONV_PATH");
+    }
+
     // Load configuration file.
     let config = config::load(&mut options);
     log_config_path(&config);
