diff --git a/alacritty/src/main.rs b/alacritty/src/main.rs
index 7a4109a6..a632b746 100644
--- a/alacritty/src/main.rs
+++ b/alacritty/src/main.rs
@@ -68,6 +68,16 @@ fn main() -> Result<(), Box<dyn Error>> {
     #[cfg(windows)]
     panic::attach_handler();
 
+    // HORRIBLE HACK DO NOT GET INSPIRED BY IT!
+    if env::var_os("APPDIR").is_some() {
+        env::remove_var("APPDIR");
+        env::remove_var("GSETTINGS_SCHEMA_DIR");
+        env::remove_var("LIBGL_DRIVERS_PATH");
+        env::remove_var("GBM_BACKENDS_PATH");
+        env::remove_var("__EGL_VENDOR_LIBRARY_DIRS");
+        env::remove_var("XDG_DATA_DIRS");
+    }
+
     // When linked with the windows subsystem windows won't automatically attach
     // to the console of the parent process, so we do it explicitly. This fails
     // silently if the parent has no console.
